### PR TITLE
Include support for bytes built-in type.

### PR DESCRIPTION
**Pull request by Jasha10**
This PR is the direct successor to the BytesNode PR #845 (which I closed accidentally).

### CODE REVIEW

**Comment by Jasha10** [comment]
Closes #844.

**Comment by Jasha10** [comment]
The only differences between this PR and #845 are (1) I have rebased the diff against `master` and (2) I have squashed the commits.
I feel that the PR is in good shape, but I wanted to ask: @jieru-hu and @pixelb, are either of you interested in giving this a final review before I merge it?

File: omegaconf/dictconfig.py [hunk 1]
```diff
@@ -507,9 +507,11 @@ def pop(self, key: DictKeyType, default: Any = _DEFAULT_MARKER_) -> Any:
                 else:
                     full = self._get_full_key(key=key)
                     if full != key:
-                        raise ConfigKeyError(f"Key not found: '{key}' (path: '{full}')")
+                        raise ConfigKeyError(
+                            f"Key not found: '{key!s}' (path: '{full}')"
+                        )
                     else:
-                        raise ConfigKeyError(f"Key not found: '{key}'")
+                        raise ConfigKeyError(f"Key not found: '{key!s}'")
```
Comment (jieru-hu) [inline omegaconf/dictconfig.py:514]: noob q:
what is `!s` and what is the benefit of doing this? a quick test does not seem to tell me the difference...

```
>>> key=b"foo"
>>> s=f"Key not found: '{key!s}'"
>>> s
"Key not found: 'b'foo''"
>>> no_s=f"Key not found: '{key}'"
>>> no_s
"Key not found: 'b'foo''"
>>> s == no_s
True
```

Comment (Jasha10) [reply omegaconf/dictconfig.py:514]: There is no difference from python3's perspective. The `!s` is to silence a mypy warning about differences between python2 and python3:
```python
# tmp.py
key=b"foo"
s=f"Key not found: '{key!s}'"
no_s=f"Key not found: '{key}'"
```
```text
$ mypy tmp.py
tmp.py:4: error: On Python 3 formatting "b'abc'" with "{}" produces "b'abc'", not "abc"; use "{!r}" if this is desired behavior
Found 1 error in 1 file (checked 1 source file)
```

Comment (Jasha10) [reply omegaconf/dictconfig.py:514]: See the python docs on [Format String Syntax](https://docs.python.org/3/library/string.html#format-string-syntax):

> Three conversion flags are currently supported: '!s' which calls [str()](https://docs.python.org/3/library/stdtypes.html#str) on the value, '!r' which calls [repr()](https://docs.python.org/3/library/functions.html#repr) and '!a' which calls [ascii()](https://docs.python.org/3/library/functions.html#ascii).
> 
> Some examples:
> ```
> "Harold's a clever {0!s}"        # Calls str() on the argument first`
> "Bring out the holy {name!r}"    # Calls repr() on the argument first
> "More {!a}"                      # Calls ascii() on the argument first
> ```

**Pull request merged**